examples/bound-vs-free.kl

```diff
@@ -0,0 +1,27 @@
+#lang "n-ary-app.kl"
+
```

Comment:
If I add `(import "quasiquote.kl")` here I get the following failure on this file and I really don't understand why.

```
Ambiguous identifier in phase p1 #[/Users/gelisam/working/haskell/klister/examples/quasiquote.kl:49.63-49.66]
                                  <#%app>
    ScopeSet {_universalScopes = fromList [Scope {scopeNum = 3, scopePurpose = "Root for phase p1"},Scope {scopeNum = 78, scopePurpose = "Module root for /Users/gelisam/working/haskell/klister/examples/quasiquote.kl"},Scope {scopeNum = 79, scopePurpose = "For declaration at siquote.kl:3.1-4.1"},Scope {scopeNum = 80, scopePurpose = "For declaration at siquote.kl:4.1-7.1"}], _phaseScopes = fromList [(Phase {phaseNum = 0},fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"}])]}
    ScopeSet {_universalScopes = fromList [Scope {scopeNum = 3, scopePurpose = "Root for phase p1"},Scope {scopeNum = 78, scopePurpose = "Module root for /Users/gelisam/working/haskell/klister/examples/quasiquote.kl"},Scope {scopeNum = 79, scopePurpose = "For declaration at siquote.kl:3.1-4.1"}], _phaseScopes = fromList [(Phase {phaseNum = 0},fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"}])]}
    ScopeSet {_universalScopes = fromList [Scope {scopeNum = 3, scopePurpose = "Root for phase p1"},Scope {scopeNum = 78, scopePurpose = "Module root for /Users/gelisam/working/haskell/klister/examples/quasiquote.kl"},Scope {scopeNum = 79, scopePurpose = "For declaration at siquote.kl:3.1-4.1"},Scope {scopeNum = 80, scopePurpose = "For declaration at siquote.kl:4.1-7.1"}], _phaseScopes = fromList [(Phase {phaseNum = 2},fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"}])]}
    ScopeSet {_universalScopes = fromList [Scope {scopeNum = 3, scopePurpose = "Root for phase p1"},Scope {scopeNum = 78, scopePurpose = "Module root for /Users/gelisam/working/haskell/klister/examples/quasiquote.kl"},Scope {scopeNum = 79, scopePurpose = "For declaration at siquote.kl:3.1-4.1"}], _phaseScopes = fromList [(Phase {phaseNum = 2},fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"}])]}
    ScopeSet {_universalScopes = fromList [Scope {scopeNum = 3, scopePurpose = "Root for phase p1"},Scope {scopeNum = 78, scopePurpose = "Module root for /Users/gelisam/working/haskell/klister/examples/quasiquote.kl"},Scope {scopeNum = 79, scopePurpose = "For declaration at siquote.kl:3.1-4.1"},Scope {scopeNum = 80, scopePurpose = "For declaration at siquote.kl:4.1-7.1"}], _phaseScopes = fromList [(Phase {phaseNum = 2},fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"}])]}
    ScopeSet {_universalScopes = fromList [Scope {scopeNum = 3, scopePurpose = "Root for phase p1"},Scope {scopeNum = 78, scopePurpose = "Module root for /Users/gelisam/working/haskell/klister/examples/quasiquote.kl"},Scope {scopeNum = 79, scopePurpose = "For declaration at siquote.kl:3.1-4.1"}], _phaseScopes = fromList [(Phase {phaseNum = 2},fromList [Scope {scopeNum = 0, scopePurpose = "Root for phase p0"}])]}
user error
```

Reply:
That is confusing! At the very least, we need to improve this message for our own sake.

The way to debug this for now is to hack up `resolve` in `Expander.hs` to show which bindings each refers to, and then to follow that back through the code to see if it's really ambiguous. As a start, we should at least show the source locations of each of the bindings that we're referring to.